presto-main/src/test/java/com/facebook/presto/operator/scalar/TestDateTimeFunctionsBase.java

```diff
@@ -1092,13 +1098,27 @@ private static long millisBetween(ReadableInstant start, ReadableInstant end)
     private SqlTime toTime(DateTime dateTime)
     {
         if (isLegacyTimestamp(session)) {
+            DateTimeZone zone = getDateTimeZone(session.getTimeZoneKey());
+            dateTime = normalizeTimeRepresentation(dateTime, new DateTime(1970, 1, 1, 0, 0, 0, zone).getMillis(), new DateTime(1970, 1, 1, 23, 59, 59, 999, zone).getMillis());
             return new SqlTime(dateTime.getMillis(), session.getTimeZoneKey());
         }
         else {
+            dateTime = normalizeTimeRepresentation(dateTime, 0, DAYS.toMillis(1) - 1);
             return new SqlTime(dateTime.getMillisOfDay());
         }
```

Comment:
I would rewrite `toTime` like this

```
    private SqlTime toTime(DateTime dateTime)
    {
        if (isLegacyTimestamp(session)) {
            dateTime = dateTime.withDate(new org.joda.time.LocalDate(1970, 1, 1));
            return new SqlTime(dateTime.getMillis(), session.getTimeZoneKey());
        }
        else {
            dateTime = dateTime.withZoneRetainFields(DateTimeZone.UTC);
            dateTime = dateTime.withDate(new org.joda.time.LocalDate(1970, 1, 1));
            return new SqlTime(dateTime.getMillis());
        }
    }

Reply:
That would change semantics. Also, my ideal rewrite would be to `[java.time.] LocalTime`, i didn't want to do this just yet.

Reply:
This doesn't change semantics. It's just another way to express that you want to clamp the timestamp to the intended range. In addition, I verified that this doesn't change the outcome of any tests.

I acknowledge that LocalDateTime would make more sense. But this change is on the right direction toward that.

Reply:
hm, i am aware my code isn't perfect, but yours isn't readable to me either. I think either way we have a problem and only going to LocalTime directly is going to fix that problem. I really prefer to keep this out of the way for this PR. What do you think?

Reply:
I am not convince on this. I would prefer to go straight to LocalTime. I view this change as minimal hack to get this working.

Reply:
The "DateTime as TIME" normalization is getting rid of in https://github.com/prestodb/presto/pull/11036